src/topology/separation.lean

```diff
@@ -59,6 +59,38 @@ end separated
 class t0_space (α : Type u) [topological_space α] : Prop :=
 (t0 : ∀ x y, x ≠ y → ∃ U:set α, is_open U ∧ (xor (x ∈ U) (y ∈ U)))
 
+theorem exists_closed_singleton_of_compact_in_closed {α : Type*} [topological_space α]
```

Comment:
The last bit of the name sounds like you have a compact subset in something closed... I would just drop the `compact` and make it
```suggestion
theorem exists_closed_singleton_of_is_closed {α : Type*} [topological_space α]
```
the rest of the assumptions are "implicit" and can be discovered by hovering etc, if they are not immediate to the user.

Reply:
I think we could put this (and the `exists_minimal_nonempty_closed_subset_of_compact`) in the `is_closed` namespace for dot notation, right?
```suggestion
theorem is_closed.exists_closed_singleton {α : Type*} [topological_space α]
```
Also, `(S : set \a)` can be made implicit, both here and there.

Reply:
Thanks for the better names.  They also fit nicely in the `is_closed` namespace.